examples/platforms/simulation/misc.c

```diff
@@ -60,6 +60,18 @@ void otPlatReset(otInstance *aInstance)
 #endif // OPENTHREAD_PLATFORM_USE_PSEUDO_RESET
 }
 
+#if OPENTHREAD_CONFIG_PLATFORM_BOOTLOADER_MODE_ENABLE
+otError otPlatResetToBootloader(otInstance *aInstance)
+{
+    otError error = OT_ERROR_NOT_CAPABLE;
+    OT_UNUSED_VARIABLE(aInstance);
+    assert(false);
```

Comment:
The return value is `OT_ERROR_NOT_CAPABLE `. Why we need to add `assert(false);`?